source/acquaman/IDEAS/IDEAS2DScanActionController.cpp

```diff
@@ -106,14 +106,7 @@ void IDEAS2DScanActionController::buildScanControllerImplementation()
     transmissionAB->setNormalizationName("Sample");
     scan_->addAnalyzedDataSource(transmissionAB);
 
-    AMXRFDetector *detector = 0;
-
-	if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ketek))
-		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("KETEK"));
-
-	else if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ge13Element) && IDEASBeamline::ideas()->ge13Element()->isConnected())
-		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("13-el Ge"));
-
+	AMXRFDetector *detector = IDEASBeamline::ideas()->xrfDetector(configuration_->fluorescenceDetector());
```

Comment:
After looking into it more, this is okay.  I was mixing up the check of whether or not there was an XRF detector at all vs. setting up the data sources properly if there are multiple XRF detectors.